misc/python/materialize/checks/multiple_partitions.py

```diff
@@ -0,0 +1,103 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+from typing import List
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+
+
+def schemas() -> str:
+    return dedent(
+        """
+       $ set keyschema={
+           "type": "record",
+           "name": "Key",
+           "fields": [
+               {"name": "key1", "type": "string"}
+           ]
+         }
+
+       $ set schema={
+           "type" : "record",
+           "name" : "test",
+           "fields" : [
+               {"name":"f1", "type":"string"}
+           ]
+         }
+    """
+    )
+
+
+class MultiplePartitions(Check):
+    """Test that adds new partitions to a Kafka source"""
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            schemas()
+            + dedent(
+                """
+                $ kafka-create-topic topic=movies-topic
+
+                $ kafka-ingest format=avro key-format=avro topic=movies-topic key-schema=${keyschema} schema=${schema} repeat=1000
+                {"key1": "A${kafka-ingest.iteration}"} {"f1": "A${kafka-ingest.iteration}"}
+
+                > CREATE CONNECTION IF NOT EXISTS kafka_conn FOR KAFKA BROKER '${testdrive.kafka-addr}';
+
+                > CREATE CONNECTION IF NOT EXISTS csr_conn FOR CONFLUENT SCHEMA REGISTRY URL '${testdrive.schema-registry-url}';
+
+                > CREATE SOURCE movies_source
+                  FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-movies-topic-${testdrive.seed}', TOPIC METADATA REFRESH INTERVAL MS 500)
+                  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn
+                  ENVELOPE UPSERT
+
+                > CREATE MATERIALIZED VIEW mv_movies AS SELECT * FROM movies_source;
+                
+                $ kafka-add-partitions topic=movies-topic total-partitions=4
+                """
+            )
+        )
+
+    def manipulate(self) -> List[Testdrive]:
+        return [
+            Testdrive(schemas() + dedent(s))
+            for s in [
+                """
+                $ kafka-ingest format=avro key-format=avro topic=movies-topic key-schema=${keyschema} schema=${schema} repeat=40
```

Comment:
The thing that should be `manipulate`-d in this test is the number of partitions. Therefore, each of the manipulate steps should be something like:
- ingest some stuff
- increase partition count
- ingest some more stuff

Reply:
Ok, I was already briefly going into this direction but wanted some feedback first because I realized that I will need to adjust the testdrive rust code to allow that (suffixes are appended to the topic name, which differ in the different functions and iterations, causing "topic not found" errors).

Reply:
Yes, I saw that. This "topic not found" message in testdrive can be thought of as a linter -- it is just a metadata check, it does not impact the rest of the logic as far as I could see.  You can just remove that check and let this be an acceptable action to take.